src/main/java/util/BogusRequest.java

```diff
@@ -0,0 +1,100 @@
+package util;
+
+import org.apache.http.client.methods.*;
+import org.apache.http.params.HttpParams;
+import org.apache.http.params.BasicHttpParams;
+import org.apache.http.Header;
+import org.apache.http.RequestLine;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.HeaderIterator;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Random;
+
+public class BogusRequest implements HttpUriRequest {
```

Comment:
This class seems needlessly complex, as you're implementing an interface just to randomize the method, but you don't care about *any* of the other methods you need to implement.

I'm sure there's a much simpler solution that you can come up with.